src/pybind/mgr/volumes/module.py

```diff
@@ -431,15 +431,13 @@ def __init__(self, *args, **kwargs):
         self.inited = False
         # for mypy
         self.max_concurrent_clones = None
-        self.lock = threading.Lock()
```

Comment:
This was introduced to address this corner case.

https://github.com/ceph/ceph/pull/36554#discussion_r468454350

Reply:
Ah, I did some research on this and my understanding of the GIL was not quite right. Python doesn't wait for calls out to C (which may block) to release the GIL; i.e. it does not serially execute code until something must block like a mutex lock or syscall. The GIL is dropped periodically (every 15ms) which forces us to consider this  possibility. Yay.

I'll drop this commit.